Sources/KlaviyoSwift/StateManagement.swift

```diff
@@ -272,6 +275,38 @@ struct KlaviyoReducer: ReducerProtocol {
             state.requestsInFlight = []
             return .none
             
+        case .enqueueEvent(let event):
+            guard case .initialized = state.initalizationState,
+                    let apiKey = state.apiKey,
+                    let anonymousId = state.anonymousId else {
+                return .none
+            }
+            let event = KlaviyoAPI.KlaviyoRequest.KlaviyoEndpoint.CreateEventPayload.Event(event: event,
+                                                                                             anonymousId: anonymousId)
+            let payload = KlaviyoAPI.KlaviyoRequest.KlaviyoEndpoint.CreateEventPayload(data: event)
+            let endpoint = KlaviyoAPI.KlaviyoRequest.KlaviyoEndpoint.createEvent(payload)
+            let request = KlaviyoAPI.KlaviyoRequest(apiKey: apiKey, endpoint: endpoint)
+            state.queue.append(request)
+            return .none
+        case .enqueueProile(let profile):
+            guard case .initialized = state.initalizationState,
+                    let apiKey = state.apiKey,
+                    let anonymousId = state.anonymousId else {
+                return .none
+            }
+            let profile = KlaviyoAPI.KlaviyoRequest.KlaviyoEndpoint.CreateProfilePayload.Profile(profile: profile,
+                                                                                                 anonymousId: anonymousId)
+            let payload = KlaviyoAPI.KlaviyoRequest.KlaviyoEndpoint.CreateProfilePayload(data: profile)
+            let endpoint = KlaviyoAPI.KlaviyoRequest.KlaviyoEndpoint.createProfile(payload)
+            let request = KlaviyoAPI.KlaviyoRequest(apiKey: apiKey, endpoint: endpoint)
+            state.queue.append(request)
+            return .none
+        case .resetProfile:
+            state.email = nil
```

Comment:
I think we also ought to clear push token in `resetProfile` ?

Reply:
So that the developer has to call `setPushToken` again on the new profile, for consistent behavior.

Reply:
I believe I addressed this...